awspub/api.py

```diff
@@ -0,0 +1,165 @@
+import logging
+import pathlib
+from typing import Dict, Iterator, List, Optional, Tuple
+
+from awspub.context import Context
+from awspub.image import Image, _ImageInfo
+from awspub.s3 import S3
+
+logger = logging.getLogger(__name__)
+
+
+def _images_grouped(
+    images: List[Tuple[str, Image, Dict[str, _ImageInfo]]], group: Optional[str]
+) -> Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]]:
+    """
+    Group the given images by name and by group
+
+    :param images: the images
+    :type images: List[Tuple[str, Image, Dict[str, _ImageInfo]]]
+    :param group: a optional group name
+    :type group: Optional[str]
+    :return: the images grouped by name and by group
+    :rtype: Tuple[Dict[str, Dict[str, str]], Dict[str, Dict[str, Dict[str, str]]]
```

Comment:
Why is this needed, it's already in the signature above.

Reply:
looks like there is https://github.com/tox-dev/sphinx-autodoc-typehints that might help here.

Reply:
Filled https://github.com/canonical/awspub/issues/56 to explore this more.